### ISSUE TITLE
build/rpms fails without rpmbuild installed, causing `make bazel-build` to fail

### ISSUE THREAD

**Issue by BenTheElder** [kubernetes/kubernetes#55437]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**: `make bazel-build` can fail with confusing errors caused by `build/rpms` (see: https://github.com/kubernetes/test-infra/issues/5429)

**What you expected to happen**: These targets should either be manual only or be hermetic. Additionally, these are not being run in CI because nothing depends on them. If they are not made manual then the CI needs to be updated to include them and needs to install `rpmbuild`.

**How to reproduce it (as minimally and precisely as possible)**:  run `make bazel-build` against kubernetes master from a linux box *without* `rpmbuild` in $PATH.

**Anything else we need to know?**:

**Environment**:
- ref: `899f7ff97d5fa5f16831b3e09c6cb9ccfdf73479`
- Bazel 0.7.0
- Linux, without `rpmbuild`

/cc @kad, @ixdy

**Comment by BenTheElder** [comment]
/assign
/assign @mikedanese, @kad, @ixdy
(please route as appropriate)

TLDR: `pkg_rpm` will eventually call `rpmbuild` which needs to be installed on the host to work, which causes builds to fail when it is not.

Depending on how we intend to support RPMs we need to either mark these targets manual or start hard depending on `rpmbuild` in more places (the `bazelbuild` image in CI, `planter` in test-infra...) and make sure this is being tested.

**Comment by mikedanese** [comment]
This is a bazelbuild issue, not a kubernetes issue.

**Comment by BenTheElder** [comment]
@mikedanese ok, but we still need to handle it. `make bazel-build` failing while CI / presubmits are green *is* a kubernetes issue.

**Comment by kad** [comment]
yes, bazel's rules depends on pre-installed tools. Ideally, to make it less dependent from host tools and produce correct packages for different distributions tools like rpmbuild should be called within either choorts or better inside docker. This way we will be able to produce correct packages e.g. for CentOS/Fedora/OpenSUSE... but that require better implementation than stock pkg_rpm.

**Comment by BenTheElder** [comment]
Agreed, what do we want to do in the meantime though? I'm ok with hard-depending on rpmbuild for the moment if that's the answer, but we will will need to:
- document the requirement somewhere
- make CI actually build these
- add rpmbuild to our images
- figure out how to deal with this when we try to finish fixing native mac bazel builds (which otherwise mostly work now)

If these don't need to be built on every build then making the targets manual would also work.

**Comment by ixdy** [comment]
I'm fine adding these tools to our bazelbuild image for now, but we should file an issue against upstream bazel because this is not a workable long-term solution.

**Comment by kad** [comment]
quick solution will be to add all needed tools to images. IMO, we do want to produce packages on every build, so we can start publishing those somewhere and be able to test them frequently.

**Comment by BenTheElder** [comment]
Ok, I also want to do something about those building locally as well though, I'm opening an issue upstream with bazel, but perhaps in the meantime we can at least note the requirement somewhere in the developer documentation.

The resulting error message is not particularly helpful so developers on systems without `rpmbuild` will be stuck when trying to `make bazel-build` on an otherwise functioning environment.

**Comment by soltysh** [comment]
My personal approach would be to require rpmbuild only for `make bazel-release` and leave `make bazel-build` produce just binaries.

**Comment by mikedanese** [comment]
We need a target that builds everything.

**Comment by BenTheElder** [comment]
This should at least give a better warning now hopefully, a patch was made upstream. I'll be adding a CI target to make sure [planter](https://github.com/kubernetes/test-infra/tree/master/planter) can bazel build soonish which will give us more signal.

**Comment by BenTheElder** [comment]
These targets are manual now, they will only build if you specifically request them.
